type-definitions/index.d.ts

```diff
@@ -1,19 +1,18 @@
 declare module "redux-persist" {
   import { Store, StoreEnhancer } from "redux";
 
-  export interface Storage {
+  export interface Storage<Result> {
     setItem(key: string, value: any, onComplete?: OnComplete<any>): Promise<any>;
-    getItem<Result>(
+    getItem(
       key: string,
       onComplete?: OnComplete<Result | string>
     ): Promise<Result | string>;
     removeItem(key: string, onComplete?: OnComplete<any>): Promise<any>;
-    getAllKeys?<Result>(
+    getAllKeys?(
       callback?: (error?: Error, keys?: string[]) => void
     ): Promise<string[]>;
-    getAllKeys?<Result>(onComplete?: OnComplete<Result>): Promise<Result>;
+    getAllKeys?(onComplete?: OnComplete<Result>): Promise<Result>;
     keys?: (...args: any[]) => any;
-    [key: string]: any; // In case Storage object has some other (private?) methods and properties.
```

Comment:
This is not needed for an interface - the concrete Storage object would be of the right type (e.g. AsyncStorage) and able to access those other methods and properties, while redux-persist does not care for the existence of those other methods and properties for it only relies on the methods/properties defined here in the interface.